src/FsAutoComplete.Core/InlayHints.fs

```diff
@@ -770,6 +774,77 @@ let tryGetExplicitTypeInfo (text: NamedText, ast: ParsedInput) (pos: Position) :
           } }
   )
 
+///<returns>
+/// List of all curried params.
+///
+/// For each curried param:
+/// * its range (including params)
+/// * range of tupled params
+///   * Note: one range when no tuple
+///
+/// ```fsharp
+/// f alpha (beta, gamma)
+/// ```
+/// ->
+/// ```fsharp
+/// [
+///   (2:7, [2:7])
+///   (8:21, [9:13; 15:20])
+/// ]
+/// ```
+///</returns>
+let private getArgRangesOfFunctionApplication (ast: ParsedInput) pos =
```

Comment:
`FSharpParseFileResults.GetAllArgumentsForFunctionApplicationAtPostion` and `getArgRangesOfFunctionApplication` are slightly different:
* `GetAllArgumentsForFunctionApplicationAtPostion`: returns a single list with all args in order (no difference between normal arg & tuple element arg)
  ```fsharp
  let f a (b,c) d = ...
  f 1 tup 7
  f 2 (3,4) 7
  ```
  ->
  ```fsharp
  [range(1); range(tup); range(7)]
  [range(2); range(3); range(4); range(7)]
  ```
  -> simple access to all args, but cannot correlate params with args
* `getArgRangesOfFunctionApplication`: returns one list with all tuple element args for each param (-> nested list). Basically same as `FSharpMemberOrFunctionOrValue.CurriedParameterGroups` just at caller side instead of definition
  ```fsharp
  let f a (b,c) d = ...
  f 1 tup 7
  f 2 (3,4) 7
  ```
  ->
  ```fsharp
  [[range(1)]; [range(tup)]; [range(7)]]
  [[range(2)]; [range(3); range(4)]; [range(8)]]
  ```
  -> difficult_ish access to all args (scattered over multiple list), but can correlate params with args

-> different behaviour & purpose

<br/>
<br/>

I don't think we have to add this to FCS.  
Unless it's used (or useful) somewhere else too (outside of FSAC/LSP), it just adds unnecessary API complexity to FCS. And having the implementation here in FSAC allows for easier/faster changes (and bug fixes)